gitpod-io/gitpod
# Support version pin for JetBrains warmup prebuild

## Description


## Related Issue(s)

Fixes ENT-261

## How to test

- Open preview env
- Pin IDE version
- Create a repository with JetBrains warmup indexing enabled like https://github.com/Gitpod-Samples/spring-petclinic
- Enable and trigger prebuild for this repo (used to cost around 15 minutes)
- Create a workspace for repo
- Check warmup logs in `/workspace/.gitpod`, it should print expected IDE version you have pinned

**Or join my org**
- [Join my org](https://hw-jb-pin.preview.gitpod-dev.com/orgs/join?inviteId=b1b5de08-d6c4-4f27-ba0c-d40303f7dc17)
- There's a [prebuild](https://hw-jb-pin.preview.gitpod-dev.com/prebuilds/5d27cafc-6e45-4b54-b090-4c874aa654ba) triggered already with pinned IntelliJ `2023.3.6`
- Start workspace with repo https://hw-jb-pin.preview.gitpod-dev.com/#https://github.com/mustard-mh/spring-petclinic
- Check warmup logs in `/workspace/.gitpod`, it should print expected IDE version `2023.3.6`

**Compatibility**
Try to start workspaces with editors, they should work like before
- Browser VS Code
- Desktop VS Code
- IntelliJ
- Pinned IntelliJ

✅ 
<img width="1394" alt="image" src="https://github.com/gitpod-io/gitpod/assets/20944364/1ccc9cd3-56f8-4b34-b2ff-4b0135bc32d7">
✅ 
|Browser VS Code| Desktop VS Code | Goland | Pinned IntelliJ
|:-:|:-:|:-:|:-:|
|<img width="1011" alt="image" src="https://github.com/gitpod-io/gitpod/assets/20944364/82bb704b-cdae-49f4-8262dc30cbc8e7af">|<img width="1718" alt="image" src="https://github.com/gitpod-io/gitpod/assets/20944364/508d2703-36d4-4b29-a988-53f5654384e7">|<img width="818" alt="image" src="https://github.com/gitpod-io/gitpod/assets/20944364/77890df3-7078-48fd-8c5f-3e18cd7fa9fb">|<img width="796" alt="image" src="https://github.com/gitpod-io/gitpod/assets/20944364/44e04131-2753-4e45-80a6-b86ad1694ab9">|




## Documentation


#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - hw-jb-pin</li>
	<li><b>🔗 URL</b> - <a href="https://hw-jb-pin.preview.gitpod-dev.com/workspaces" target="_blank">hw-jb-pin.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - hw-jb-pin-gha.26575</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-hw-jb-pin%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-dev-preview" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

</details>

<details>
<summary>Installer</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
</details>

/hold
